codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -757,7 +779,7 @@ void fireChildRead(Http2Frame frame) {
             assert eventLoop().inEventLoop();
             if (!isActive()) {
                 ReferenceCountUtil.release(frame);
-            } else if (readInProgress) {
+            } else if (readStatus != ReadStatus.IDLE) {
                 // If readInProgress there cannot be anything in the queue, otherwise we would have drained it from the
```

Comment:
"If a read is in progress or has been requested, there cannot be anything in the queue" (etc)